### PR TITLE
DowngradePropertyPromotionRector: prevent unnecessary work

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

File: rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php [hunk 1]
```diff
@@ -79,17 +79,18 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        $oldComments = $this->getOldComments($node);
-        $promotedParams = $this->resolvePromotedParams($node);
+        $constructorClassMethod = $node->getMethod(MethodName::CONSTRUCT);
+        if (! $constructorClassMethod instanceof ClassMethod) {
+            return [];
+        }
```
Comment (staabm) [inline rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php:85]: moved this condition from the inner methods up to the top, as it indicates an early exit condition

File: rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php [hunk 2]
```diff
@@ -79,17 +79,18 @@ public function getNodeTypes(): array
...
+        $constructorClassMethod = $node->getMethod(MethodName::CONSTRUCT);
+        if (! $constructorClassMethod instanceof ClassMethod) {
             return null;
         }
 
-        /** @var ClassMethod $constructClassMethod */
-        $constructClassMethod = $node->getMethod(MethodName::CONSTRUCT);
-
-        $properties = $this->resolvePropertiesFromPromotedParams($constructClassMethod, $promotedParams, $node);
+        $oldComments = $this->getOldComments($constructorClassMethod);
```
Comment (samsonasik) [inline rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php:87]: `oldComments` definition can be moved below if return early

Comment (staabm) [reply rules/DowngradePhp80/Rector/Class_/DowngradePropertyPromotionRector.php:87]: comments will be modified in `resolvePromotedParams()` therefore the "old" comments need to be stored before

**Comment by samsonasik** [comment]
Thank you @staabm

**Pull request merged**